x/stakeibc/keeper/lsm.go

```diff
@@ -125,6 +126,10 @@ func (k Keeper) GetHostZoneFromLSMTokenPath(ctx sdk.Context, path string) (types
 	// Confirm the channel is from one of Stride's supported host zones
 	for _, hostZone := range k.GetAllHostZone(ctx) {
 		if hostZone.TransferChannelId == channelId {
+			if !hostZone.LsmLiquidStakeEnabled {
```

Comment:
Will this error bubble up through the function calls and ultimately prevent stride from accepting the LSM token in the first place if it is from a host zone we don't have marked as enabled?

Reply:
I guess that asks the question, if we are getting an otherwise valid LSM NFT from a chain we think is not enabled... are we using the flag to really turn off LSM capabilities from some chains at that point for some strategic reason?

Reply:
It's a good question. 

You would expect tokens from chains who do not have LSM enabled to fail `IsValidIBCPath`. 

But I don't think we can be sure. Maybe a host zone with LSM disabled gives a rogue smart contract access to mint tokens of denoms that satisfy `IsValidIBCPath`. 

I view this line as a sanity check for that case. 

Maybe we should do this check before `IsValidIBCPath`, wdyt @sampocs?

Reply:
`IsValidIBCPath` is just a regex [expression](https://github.com/Stride-Labs/stride/pull/777/files/4017b18573d4fd13b4cb0616bfc5ff309f197ad7#diff-9d3cc274196539a649a91697864567e3ce6340cd5ec5d5543cb94c3bd3e0fed7R27). 

We wouldn't want to accept valid LSM NFTs for a non-LSM enabled host zone cause we would essentially be holding their token and doing nothing with it. However, this would only occur in the downtime between when a host zone upgrades to include LSM and when enable it on our end

Reply:
So related to my other question below, we only foresee moving from not yet enabled to enabled.  There is no reason we would ever want to move the other direction where even though in fact that chain does have the LSM module, we as stride want to turn off the LSM ability for strategic reasons -- maybe their validators are unhappy about LSM for one reason or another or we don't trust them after we somehow get a whole bunch of LSM tokens stuck which won't detokenize or something else weird.  This isn't foreseen as a feature kill switch, only intended in the one way upgrade process